Plugins/DotNET/DotNET.cpp

```diff
@@ -122,50 +154,85 @@ static bool InitThunks()
     return true;
 }
 
-static void DotNET()
+void LoadNetCore(const std::string& assembly)
 {
-    // If the initial lib loads failed, we probably don't have .NET installed.
-    if (!InitThunks())
-        return;
-
-    auto assembly   = Config::Get<std::string>("ASSEMBLY");
-    auto entrypoint = Config::Get<std::string>("ENTRYPOINT", "NWN.Internal");
-
-    if (!assembly.has_value())
-    {
-        LOG_ERROR("NWNX_DOTNET_ASSEMBLY not specified. DotNET plugin will not be loaded.");
-        LOG_ERROR("If you're not using the .NET plugin, you can disable this message with 'NWNX_DOTNET_SKIP=y'");
-        return;
-    }
-
-    // Load .NET Core
     hostfxr_handle cxt = nullptr;
-    auto runtimeconfig = *assembly + ".runtimeconfig.json";
-    int rc = hostfxr_initialize_for_runtime_config(runtimeconfig.c_str(), nullptr, &cxt);
-    if (rc != 0 || cxt == nullptr)
-        LOG_FATAL("Unable to load runtime config '%s'; rc=0x%x", runtimeconfig, rc);
+
+    auto runtimeConfig = assembly + ".runtimeConfig.json";
+    int returnCode = hostfxr_initialize_for_runtime_config(runtimeConfig.c_str(), nullptr, &cxt);
+    if (returnCode != 0 || cxt == nullptr)
+        LOG_FATAL("Unable to load runtime config '%s'; returnCode=0x%x", runtimeConfig, returnCode);
 
     // Get the load assembly function pointer
-    load_assembly_and_get_function_pointer_fn load_assembly_and_get_function_pointer = nullptr;
-    rc = hostfxr_get_runtime_delegate(cxt, hdt_load_assembly_and_get_function_pointer, (void**)&load_assembly_and_get_function_pointer);
-    if (rc != 0 || load_assembly_and_get_function_pointer == nullptr)
-        LOG_FATAL("Unable to get load_assembly_and_get_function_pointer; rc=0x%x", rc);
+    load_assembly_and_get_function_pointer = nullptr;
+    returnCode = hostfxr_get_runtime_delegate(cxt, hdt_load_assembly_and_get_function_pointer, (void**)&load_assembly_and_get_function_pointer);
+    if (returnCode != 0 || load_assembly_and_get_function_pointer == nullptr)
+        LOG_FATAL("Unable to get load_assembly_and_get_function_pointer; returnCode=0x%x", returnCode);
 
     hostfxr_close(cxt);
+}
 
+void CoreMessageHandler(const std::vector<std::string> &message)
+{
+    const std::string& messageType = message[0];
 
-    component_entry_point_fn bootstrap = nullptr;
-    auto dll = *assembly + ".dll";
-    auto full_ep = entrypoint + ", " + assembly->substr(assembly->find_last_of("/\\") + 1);
-    rc = load_assembly_and_get_function_pointer(dll.c_str(), full_ep.c_str(),
-                                                "Bootstrap", nullptr, nullptr, (void**)&bootstrap);
-    if (rc != 0 || bootstrap == nullptr)
-        LOG_FATAL("Unable to get %s.Bootstrap() function: dll='%s'; rc=0x%x", full_ep, dll, rc);
+    if (messageType == "ON_NWNX_LOADED")
+    {
+        Bootstrap();
+    }
```

Comment:
Bootstrapping was moved to the end of NWNX init to allow for all enabled NWNX plugins to be loaded, in-case C# code attempts to use exports from other plugins.